tonic-build/src/prost.rs

```diff
@@ -0,0 +1,191 @@
+use super::schema;
+use proc_macro2::TokenStream;
+use prost_build::{Config, Method, Service};
+use quote::ToTokens;
+use std::path::{Path, PathBuf};
+
+impl<'a> schema::Commentable<'a> for Service {
+    type Comment = String;
+    type CommentIterator = std::slice::Iter<'a, String>;
+
+    fn comment(&'a self) -> Self::CommentIterator {
+        self.comments.leading.iter()
+    }
+}
+
+pub(crate) struct ProstContext {
+    pub(crate) proto_path: String,
+}
+
+impl schema::Context for ProstContext {
+    const CODEC_NAME: &'static str = "ProstCodec";
```

Comment:
so we probably want to make this non-const so that users can set it via a builder fn possibly. This path to this codec may not always be static.